src/Control/Effect/Simple.hs

```diff
@@ -0,0 +1,47 @@
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{- | An effect that promotes a simple higher-order function into a first-order effect, trading efficiency for simplicity.
+
+For more information about the tradeoffs associated with this formulation of effects, please see "Control.Carrier.Simple".
+
+Predefined carriers:
+
+* "Control.Carrier.Simple.SimpleC"
+
+@since 1.2.0.0
+-}
+
+module Control.Effect.Simple
```

Comment:
This isn't _really_ an effect; maybe we should move this into `Control.Algebra.Simple` or something?